src/python/pants/engine/target.py

```diff
@@ -26,6 +26,10 @@
 from pants.util.meta import frozen_after_init
 from pants.util.ordered_set import FrozenOrderedSet
 
+# -----------------------------------------------------------------------------------------------
+# Core Field abstractions
+# -----------------------------------------------------------------------------------------------
```

Comment:
`target.py` has grown pretty large. But, I'd prefer we keep it all in one file for now because it's all related and it greatly improves discoverability in call sites, imo - you don't have to guess which file something is defined in.

These section headers are meant to help readers to quickly find things.